File: app/src/main/java/org/wen/gank/tools/Converters.java
package org.wen.gank.tools;

import android.arch.persistence.room.TypeConverter;
import android.support.annotation.Keep;

import com.squareup.moshi.JsonAdapter;
import com.squareup.moshi.Moshi;
import com.squareup.moshi.Types;

import java.io.IOException;
import java.util.List;

@Keep
public class Converters {

    static JsonAdapter<List<String>> sJsonAdapter = new Moshi.Builder().build()
            .adapter(Types.newParameterizedType(List.class, String.class));

    @TypeConverter
    public static List<String> listFromString(String value) {
        try {
            return value == null ? null : sJsonAdapter.fromJson(value);
        } catch (IOException e) {
            e.printStackTrace();
            return null;
        }
    }

    @TypeConverter
    public static String listToString(List<String> images) {
        return images == null ? null : sJsonAdapter.toJson(images);
    }

}

File: app/src/main/java/org/wen/gank/tools/Converters.kt
package org.wen.gank.tools

import androidx.annotation.Keep
import androidx.room.TypeConverter
import com.squareup.moshi.Moshi
import com.squareup.moshi.Types
import java.io.IOException

@Keep
class Converters {

  private var sJsonAdapter = Moshi.Builder().build()
      .adapter<List<String>>(Types.newParameterizedType(List::class.java, String::class.java))

  @TypeConverter
  fun listFromString(value: String?): List<String>? {
    try {
      return if (value == null) null else sJsonAdapter.fromJson(value)
    } catch (e: IOException) {
      e.printStackTrace()
    }
    return null
  }

  @TypeConverter
  fun listToString(images: List<String>?): String? {
    return if (images == null) null else sJsonAdapter.toJson(images)
  }

}
